geopandas/tools/tests/test_sjoin.py

```diff
@@ -82,6 +82,23 @@ def test_sjoin_values(self):
         df = sjoin(self.polydf, self.pointdf, how='left')
         self.assertEquals(df.shape, (12,8))
 
+    def test_no_overlapping_geometry(self):
+        # Note: these tests are for correctly returning GeoDataFrame
+        # when result of the join is empty
+
+        subset_pointdf = self.pointdf.iloc[17:]
+
+        df_inner = sjoin(subset_pointdf, self.polydf, how='inner')
+        self.assertEquals(df_inner.shape, (0,8))
```

Comment:
You are still testing the shape and the fact that there are no values in a certain column. But I would like to see validation of the actual result. So something like

```
expected = GeoDataFrame(...)
assert_frame_equal(df_left, expected)
```